### PR TITLE
adding parallel rebalancing 

### LINKED ISSUE #1081: Doc changes for parallel rebalancing

### PR DESCRIPTION
**Pull request by arianapadilla**
fixes #1081 

@emelsimsek I documented the GUC but the job infrastructure tables were not in the docs. Do you think it's ok if we just document what we have right now?

### CODE REVIEW

**Comment by jonels-msft** [comment]
Just noticed the issue includes a nice code sample.

We could include the example in the new [cluster management section](https://citus-doc-pr-parallelre-pa9vfb.herokuapp.com/admin_guide/cluster_management.html#parallel-rebalancing).

```sql
ALTER SYSTEM SET citus.max_background_task_executors_per_node = 2;
SELECT pg_reload_conf();

SELECT citus_rebalance_start();
```

That helps tie it together for people.

**Comment by emelsimsek** [comment]
> fixes #1081
> 
> @emelsimsek I documented the GUC but the job infrastructure tables were not in the docs. Do you think it's ok if we just document what we have right now?

I agree that we can leave out the implementation details including the pg_dist_task* tables out of the user facing documentation.

**Comment by emelsimsek** [comment]
> Just noticed the issue includes a nice code sample.
> 
> We could include the example in the new [cluster management section](https://citus-doc-pr-parallelre-pa9vfb.herokuapp.com/admin_guide/cluster_management.html#parallel-rebalancing).
> 
> ```sql
> ALTER SYSTEM SET citus.max_background_task_executors_per_node = 2;
> SELECT pg_reload_conf();
> 
> SELECT citus_rebalance_start();
> ```
> 
> That helps tie it together for people.

Yes, this could be included where we talk about rebalancing in cluster management.

File: develop/api_guc.rst [hunk 1]
```diff
@@ -17,6 +17,17 @@ As an example you can update a setting with:
 General configuration
 ---------------------------------------
 
+.. _max_background_task_executors_per_node:
+
+citus.max_background_task_executors_per_node (integer)
+$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+Determines how many background tasks can be executed in parallel at a given time. For instance, these tasks are for shard moves from/to a node.
```
Comment (JelteF) [inline develop/api_guc.rst:25]:
```suggestion
Determines how many background tasks can be executed in parallel at a given time. For instance, these tasks are for shard moves from/to a node. When increasing value this you'll often also want to increase :code:`citus.max_background_task_executors` and :code:`max_worker_processes`.
```

**Pull request merged**